src/app/BreedProcessor.ts

```diff
@@ -144,4 +149,44 @@ export class BreedProcessor {
     );
     return min([herdMaxGrow, bankContains]) as number;
   }
+
+  private reduceGain(
+    predator: Predator,
+    animalsGain: [AnimalNames, number][],
+  ): void {
+    if (predator.theName === AnimalNames.WOLF) {
+      const horseIndex = this.getAnimalIndex(
+        animalsGain,
+        AnimalNames.HORSE,
+      );
+      if (horseIndex === -1) {
+        animalsGain.splice(0);
+      } else {
+        animalsGain.splice(0, horseIndex);
+        animalsGain.splice(1, animalsGain.length);
+      }
+    }
+    if (predator.theName === AnimalNames.FOX) {
+      const rabbitIndex = this.getAnimalIndex(
+        animalsGain,
+        AnimalNames.RABBIT,
+      );
+      if (rabbitIndex !== -1) {
+        animalsGain.splice(rabbitIndex, 1);
+      }
+    }
+  }
+
+  private getAnimalIndex(
+    animalsGain: [AnimalNames, number][],
+    animal: AnimalNames,
+  ): number {
+    const animalCount = animalsGain
+      .filter(([animalName]) => animal === animalName)
+      .map(([, count]) => count)
+      .pop();
```

Comment:
Can you explain why does it need to be popped?

Reply:
It is necessary to find index in array. I don't know any way to mock one part of tupple.